### PR TITLE
app-backup/restic: New package

### PR DESCRIPTION
**Pull request by stkw0**
related gentoo bug: https://bugs.gentoo.org/show_bug.cgi?id=607630

### CODE REVIEW

File: app-backup/restic/metadata.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+<pkgmetadata>
+	<maintainer type="person">
+		<email>davidroman96@gmail.com</email>
+		<name>David Roman</name>
+	</maintainer>
+	<maintainer type="project">
+		<email>proxy-maint@gentoo.org</email>
+		<name>Proxy Maintainers</name>
+	</maintainer>
+</pkgmetadata>
```
Comment (gktrk) [inline app-backup/restic/metadata.xml]: Please insert a newline at the end here :)

File: app-backup/restic/restic-0.4.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+DESCRIPTION="restic is a backup program that is fast, efficient and secure"
+HOMEPAGE="https://restic.github.io/"
+SRC_URI="https://github.com/restic/restic/archive/v${PV}.tar.gz"
```
Comment (gktrk) [inline app-backup/restic/restic-0.4.0.ebuild]: For simplicity, let's [rename](https://devmanual.gentoo.org/ebuild-writing/variables/index.html#renaming-sources) this tarball to `${P}.tar.gz`. i.e.
```
SRC_URI="https://github.com/restic/restic/archive/v${PV}.tar.gz -> ${P}.tar.gz"
```

File: app-backup/restic/restic-0.4.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+LICENSE="BSD-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86 ~arm"
```
Comment (gktrk) [inline app-backup/restic/restic-0.4.0.ebuild]: Have you personally tested this package on all these arches? despite what upstream claims, we like doing our own testing before adding arches to `KEYWORDS`.

File: app-backup/restic/restic-0.4.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+IUSE=""
```
Comment (gktrk) [inline app-backup/restic/restic-0.4.0.ebuild]: Empty `IUSE` can be removed entirely

File: app-backup/restic/restic-0.4.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+DEPEND="dev-lang/go
+		sys-fs/fuse"
```
Comment (gktrk) [inline app-backup/restic/restic-0.4.0.ebuild]: Nitpick: This should be single level indentation, not two

**Comment by gktrk** [comment]
Hi, I had some time to hack this a bit more. Here's what I did:

- got rid of build.go and used go build directly. it only required adjusting GOPATH properly.
- enabled verbose compilation messages etc.
- disabled passing the '-s' linker flag so the binary isn't pre-stripped
- passing the `-X main.version=${PV}` makes the sed trick obsolete
- used `einstalldocs` to install the documentation files
- tests do not work properly without adjusting GOPATH, so fixed that as well.

Here's the diff:
```
 app-backup/restic/restic-0.4.0.ebuild | 23 +++++++++++++++++++----
 1 file changed, 19 insertions(+), 4 deletions(-)

diff --git a/app-backup/restic/restic-0.4.0.ebuild b/app-backup/restic/restic-0.4.0.ebuild
index 34589d2..c3e99fe 100644
--- a/app-backup/restic/restic-0.4.0.ebuild
+++ b/app-backup/restic/restic-0.4.0.ebuild
@@ -11,16 +11,31 @@ SRC_URI="https://github.com/restic/restic/archive/v${PV}.tar.gz -> ${P}.tar.gz"
 LICENSE="BSD-2"
 SLOT="0"
 KEYWORDS="~amd64 ~x86"
+IUSE="test"
 
-DEPEND="dev-lang/go"
+DEPEND="dev-lang/go
+	test? ( sys-fs/fuse )"
 
 RDEPEND="sys-fs/fuse"
 
-src_prepare() {
-	sed -i 's/versionGit := getVersionFromGit()/versionGit := ""/g' "${S}/build.go"
-	default
+src_compile() {
+	local mygoargs=(
+		-v
+		-work
+		-x
+		-tags release
+		-ldflags "-w -X main.version=${PV}"
+		-o "${S}"/restic cmds/restic
+	)
+
+	GOPATH="${S}:${S}/vendor" go build "${mygoargs[@]}" || die
 }
 
 src_install() {
 	dobin restic
+	einstalldocs
+}
+
+src_test() {
+	GOPATH="${S}:${S}/vendor" go test -v -work -x restic/... cmds/... || die
 }
```

Here's the full ebuild: https://dev.gentoo.org/~gokturk/proxy-maint/pr3876/restic-0.4.0.ebuild

Let me know what you think and if you have any questions.

**Comment by robbat2** [comment]
@stkw0 @gktrk Please ensure you install all of the documentation as well, not just one file. `*.md` and `doc/*`

File: app-backup/restic/metadata.xml [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
```
Comment (SoapGentoo) [inline app-backup/restic/metadata.xml]: empty line, remove please

File: app-backup/restic/restic-0.4.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+src_install() {
+	dobin restic
+	einstalldocs
+}
+
+src_test() {
+	GOPATH="${S}:${S}/vendor" go test -v -work -x restic/... cmds/... || die
+}
```
Comment (SoapGentoo) [inline app-backup/restic/restic-0.4.0.ebuild]: put `src_test` before `src_install`

**Comment by stkw0** [comment]
@robbat2 @gktrk Thanks alot for your comments and time. I hope it works "well" now. 

PD: Inside the doc directory there were misc files that are not documentation, so I didn't include those

**Comment by gktrk** [comment]
Merged in https://github.com/gentoo/gentoo/compare/68c0da5074...052ca64684. Thanks!

**Pull request closed without merging**